vimmo/db/test_weekly_update.py

```diff
@@ -0,0 +1,78 @@
+from db import Database
+import random
+from weekly_update import main
+
+'''
+This script adds in, removes and changes the confidence of some genes in panels 3 and 9, then calls the main function
+from test_weekly_update.py to contact the PanelApp API and update the database, including changing latest version
+numbers, latest genes and confidences (in panel_genes) and archiving any previous versions (in panel_genes_archive)
+
+'''
+
+
+def update_tester(cursor):
+
+    '''
+    Function adds in, deletes and changes the confidence of genes in the panel_genes table, so the weekly update
+    and archive script can be tested
+    '''
+
+    cursor.execute(
+        '''
+        delete from panel_genes where Panel_ID = 3 and HGNC_ID = 'HGNC:2186'
+        '''
+    )
+
+    cursor.execute(
+        '''
+        insert into panel_genes (Panel_ID, HGNC_ID, Confidence) VALUES (3, 'HGNC:44444444', 1)
+        '''
+    )
+
+    cursor.execute(
+        '''
+        Update panel_genes set Confidence = 2 where Panel_ID = 3 and HGNC_ID = 'HGNC:2200'
+        '''
+    )
+
+    cursor.execute(
+        '''
+        delete from panel_genes where Panel_ID = 9 and HGNC_ID = 'HGNC:3424'
+        '''
+    )
+
+    cursor.execute(
+        '''
+        insert into panel_genes (Panel_ID, HGNC_ID, Confidence) VALUES (9, 'HGNC:44444444', 1)
+        '''
+    )
+
+    cursor.execute(
+        '''
+        Update panel_genes set Confidence = 2 where Panel_ID = 9 and HGNC_ID = 'HGNC:5211'
+        '''
+    )
+
+    random_version = round(random.uniform(0.1, 1.0), 3)
+    cursor.execute(
+        '''
+        update panel set Version = ? where Panel_ID = 3
+        ''',
+        (random_version,)
+    )
+
+    cursor.execute(
+        '''
+        update panel set Version = ? where Panel_ID = 9
+        ''',
+        (random_version,)
+    )
+
+    db.conn.commit()
+
```

Comment:
please close the connection to db after the functionality is done.

Reply:
done - any idea if it is better practice to commit and close connection in a function itself or when calling the functions in main() ?

Reply:
It depends on how you connect to the database. If you connect it once in different isolated function e.g, 
`def connection ():
    return connection = connect to db`
and this function runs in an isolated manner, so not being called in an another function to create a connection, then closing it explicitly at the end of everything creating and invoking a close function separately. However, if you are creating the connection in another function specifically then its a good idea to close it in that function aswell before it returns a value.  Although there might probably be better practices that I am not aware of.